hs_core/views/utils.py

```diff
@@ -559,84 +559,125 @@ def get_metadata_contenttypes():
-                                                                         via_group=True)
+    viewable_resources = user.uaccess.get_resources_with_explicit_access(PrivilegeCodes.VIEW, via_group=True)
+
     # remove obsoleted resources from the viewable_resources
-    viewable_resources = viewable_resources.exclude(object_id__in=Relation.objects.filter(
-        type='isReplacedBy').values('object_id'))
+    viewable_resources = viewable_resources.exclude(
+        object_id__in=Relation.objects.filter(type='isReplacedBy').values('object_id'))
 
     discovered_resources = user.ulabels.my_resources
 
-    # join all queryset objects
-    resource_collection = owned_resources.distinct() | \
-        editable_resources.distinct() | \
-        viewable_resources.distinct() | \
-        discovered_resources.distinct()
-
-    if annotate:  # When used in the My Resources page, annotate for speed
-        labeled_resources = user.ulabels.labeled_resources
-        favorite_resources = user.ulabels.favorited_resources
-        resource_collection = resource_collection.annotate(
-            owned=Case(When(short_id__in=owned_resources.values_list('short_id', flat=True),
-                            then=Value(True, BooleanField()))))
-
-        resource_collection = resource_collection.annotate(
-            editable=Case(When(short_id__in=editable_resources.values_list('short_id', flat=True),
-                          then=Value(True, BooleanField()))))
+    favorite_resources = user.ulabels.favorited_resources
 
-        resource_collection = resource_collection.annotate(
-            viewable=Case(When(short_id__in=viewable_resources.values_list('short_id', flat=True),
-                               then=Value(True, BooleanField()))))
+    return {
+        'favorites': favorite_resources.count(),
+        'ownedCount': owned_resources.count(),
+        'addedCount': discovered_resources.count(),
+        'sharedCount': viewable_resources.count() + editable_resources.count()
+    }
 
-        resource_collection = resource_collection.annotate(
-            discovered=Case(When(short_id__in=discovered_resources.values_list('short_id', flat=True),
-                            then=Value(True, BooleanField()))))
 
-        resource_collection = resource_collection.annotate(
-            is_favorite=Case(When(short_id__in=favorite_resources.values_list('short_id', flat=True),
-                                  then=Value(True, BooleanField()))))
+def get_my_resources_list(user, annotate=False, filter=None, **kwargs):
+    """
+    Gets a QuerySet object for listing resources that belong to a given user.
+    :param user: an instance of User - user who wants to see his/her resources
+    :param annotate: whether to annotate for my resources page listing.
+    :return: an instance of QuerySet of resources
+    """
+    resource_collection = BaseResource.objects.none()
```

Comment:
I think we need these `.objects.none()` assignments because otherwise we get errors when trying to union the querysets later.

Example: `filters=['owned']` 
if we don't have
` editable_resources = BaseResource.objects.none()`
then when we do
>resource_collection = owned_resources.distinct() | \
            editable_resources.distinct() | \
            viewable_resources.distinct() | \
            discovered_resources.distinct()
I think we get an error because ` editable_resources` is not defined